pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -343,6 +343,8 @@ public static class Broker {
     public static final String CONFIG_OF_ENABLE_PARTITION_METADATA_MANAGER =
         "pinot.broker.enable.partition.metadata.manager";
     public static final boolean DEFAULT_ENABLE_PARTITION_METADATA_MANAGER = false;
+    public static final String IMPLICIT_COLOCATE_JOIN = "pinot.broker.multistage.implicit.colocate";
```

Comment:
The added rule is to auto-fill the missing info in the table hint. Should we make the config more explicit about that? This rule will be applied even without `JOIN`, and leaf stage worker will process each partition in parallel

Reply:
> leaf stage worker will process each partition in parallel

By default we set parallelism to 1, which AFAIU was the default. Therefore this should not change, am I wrong?

Reply:
I've changed that to do not provide a value for parallelism. Anyway, you are right about the naming. Do you have a suggestion for a new and more precise property name?

Reply:
Remember to change this